site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,694 @@
-  short: Dieser Bot kann dir bei Dingen helfen.
-  long: >
-    Dieser Bot kann dir bei Dingen helfen.
-    Zum Beispiel…
+```fluent
+start = Hallo, wie kann ich dir helfen? /help
+help = 
+    Schick eine Textnachricht, die ich für dich fett schreiben soll.
+    Du kannst mit dem Befehl /language die Spache ändern.
 ```
 
-As you can see the same keys are existing in both files.
-When we now want to send a message to a user with have to use the key.
+Using the translations in your bot:
 
 ```ts
 bot.command("start", async (ctx) => {
-  await ctx.reply(ctx.i18n.t("hey"));
+  await ctx.reply(ctx.t("start"));
+});
+
+bot.command("help", async (ctx) => {
+  await ctx.reply(ctx.t("help"));
 });
 ```
 
-Depending on the user language the Bot will now reply with a different text.
+Whenever a message is sent the locale of context is used. This could be your specific language negotiator or just `ctx.from.language_code`
+So as a result different users with different locales will be able to read the messages in their languages.
+
+## Usage
+
+The plugin chooses the language in use by different ways. One of them is the default `ctx.from.language_code` provided by the telegram client of the user.
+If your bot has a command for changing the language it is better to use the language chosen by the user.
+Based on these different language selection approaches the i18n plugin can be configured mainly in two different ways: [With sessions](#with-sessions) or [Without sessions](#without-sessions).
+
+### Without Sessions
 
-But we need to set up the i18n support first as the bot needs to know where the actual text is coming from.
+This is easier to set up and use than using with sessions. Its main drawback is that you can’t save the chosen language of the user.
+Though technically, you can just implement a [custom locale negotiator](#custom-locale-negotiation) to achieve that.
+
+The locale for the user is selected from `ctx.from.language_code`, which is provided by the telegram client of the user.
+But if you don't have a translation of that language, the default language will be used.
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+import { Bot, Context } from "grammy";
+import { I18n, I18nContextFlavor } from "@grammyjs/i18n";
+
+// For full TypeScript support and proper auto-completion
+// extend the Context with I18n's Context Flavor.
+type MyContext = Context & I18nContextFlavor;
+
+// Create a bot as you normally would.
+// Remember to extend the context.
+const bot = new Bot<MyContext>(""); // <-- put your bot token here (https://t.me/BotFather)
+
+// Create an `I18n` instance.
+// Continue reading to find out how to configure the instance.
+const i18n = new I18n({
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
+  directory: "locales",
+});
+
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n.middleware());
```

Comment:
I have asked this before but I don't remember—why is this preferred over `bot.use(i18n)`?

Reply:
Sorry about that. iirc, you asked about this, but that wasn't working back then because there were some type issues. So, I managed to make it work in the plugin code, but forgot to update here. Anyway, I've udpated them now :) Thanks for the reminder.